### PR TITLE
Fix containers.graph sav/get_weights

### PR DESCRIPTION
**Pull request by tzachar**
As the graph container was not using each individual layer's get/set
weights, but rather the super class layer.get_weights, which works on
self.params(), it was missing some weights in the process, i.e., the
BatchNormalizationLayer has custom get_weights which allows to save the
running mean/std. However, these running computations are not added to
BatchNormalizationLayer.params(), resulting in losing these weights
after serializing a graph model utilizing a BatchNormalizationLayer.

Fixed to use each node's get/set weights.

### CODE REVIEW

File: keras/layers/containers.py [hunk 1]
```diff
@@ -316,3 +316,15 @@ def get_config(self):
 
     def count_params(self):
         return sum([layer.count_params() for layer in self.nodes.values()])
+
+    def get_weights(self):
+        weights = []
+        for layer in self.nodes.values():
```
Comment (fchollet) [inline keras/layers/containers.py]: This is fine because `nodes` is an OrderedDict. It would be somewhat unsafe for a dict. Maybe that would warrant a code comment.

**Comment by tzachar** [comment]
10x
